Sprint 4

AleksandrMenshchikov opened the pull request.

AleksandrMenshchikov committed:
modified:   package.json

AleksandrMenshchikov committed:
modified:   package.json

AleksandrMenshchikov committed:
modified:   app.ts

AleksandrMenshchikov committed:
renamed:    app.ts -> app.js
	new file:   babel.config.js
	modified:   package-lock.json
	modified:   package.json
	renamed:    src/index.html -> public/index.html
	modified:   src/components/avatar/avatar.hbs
	modified:   tsconfig.json
	new file:   webpack.config.js

AleksandrMenshchikov committed:
modified:   src/index.ts

AleksandrMenshchikov committed:
deleted:    src/components/avatar/avatar.css
	deleted:    src/components/avatar/avatar.hbs
	deleted:    src/components/avatar/avatar.ts
	deleted:    src/components/avatar/index.ts
	modified:   src/components/button-avatar/button-avatar.css
	modified:   src/components/button-avatar/button-avatar.hbs
	modified:   src/components/button-avatar/button-avatar.ts
	modified:   src/components/form-settings/form-settings.ts
	deleted:    src/hoc/withAvatar.ts
	new file:   src/hoc/withButtonAvatar.ts

AleksandrMenshchikov committed:
modified:   README.md

AleksandrMenshchikov committed:
new file:   .dockerignore
	modified:   .gitignore
	modified:   .npmrc
	new file:   Dockerfile
	modified:   package-lock.json

AleksandrMenshchikov committed:
modified:   README.md

AleksandrMenshchikov committed:
modified:   package-lock.json

AleksandrMenshchikov committed:
modified:   README.md

AleksandrMenshchikov committed:
modified:   README.md

AleksandrMenshchikov committed:
modified:   README.md

AleksandrMenshchikov committed:
modified:   README.md

AleksandrMenshchikov committed:
deleted:    .parcelrc

AleksandrMenshchikov committed:
modified:   .eslintignore

AleksandrMenshchikov committed:
deleted:    .postcssrc
	modified:   package-lock.json
	modified:   package.json
	new file:   postcss.config.js
	modified:   src/index.css

gennady-bars commented on package.json:
```diff
@@ -2,15 +2,14 @@
-    "dev": "rm -rf dist && parcel src/index.html --no-cache --open",
-    "build": "rm -rf dist && npm run lint && parcel build src/index.html --no-cache --no-content-hash --no-scope-hoist --no-optimize",
-    "start": "node app.ts",
-    "prepare": "husky install",
```
В чек-листе есть пункт: Тестами покрыты как минимум шаблонизатор (если делаете свою реализацию), **роутер**, **компонент**, **модуль отправки запросов**. Скрин из чек-листа https://disk.yandex.ru/i/fu8v7hdHCsuluw

AleksandrMenshchikov replied:
Исправил.

gennady-bars commented on package.json:
```diff
@@ -2,15 +2,14 @@
+    "start": "node app.js",
+    "lint:js": "./node_modules/.bin/eslint .",
+    "lint:css": "./node_modules/.bin/stylelint \"**/*.css\"",
+    "lint": "npm run lint:js && npm run lint:css"
```
**Можно лучше**

Для линтинга на прекоммите можно воспользоваться библиотекой [lint-staged](https://www.npmjs.com/package/lint-staged) . Она позволяет осуществлять линтинг только тех файлов, которые **изменились**

gennady-bars commented on webpack.config.js:
```diff
@@ -0,0 +1,60 @@
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+
+module.exports = {
```
**Можно лучше**

`webpack.config.js`  хорошо бы разбить на несколько частей - `development` и `production`. Так же будет еще файл с общими частями для обеих сборок. Объединить конфиги можно с помощью библиотеки `webpack-merge`

Часть функционала, которая необходима для `production` сборки, не требуется в `development` сборке. Эта работа просто будет замедлять сборку проекта.

AleksandrMenshchikov replied:
Исправил.

gennady-bars commented on webpack.config.js:
```diff
@@ -0,0 +1,60 @@
+    new CleanWebpackPlugin(),
+    new MiniCssExtractPlugin(),
+  ],
+};
```
**Можно лучше**

лучше всего передавать тип пропсов в виде дженерика в `Block`, чтобы можно было конкретный тип пропсов всегда иметь под каждый отдельный экземпляр класса `Block`. Будет очень универсально 

```js
abstract class Block<Props extends Record<string, any> = unknown> {

```

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная



* `Readme` хорошо оформлен
* Проект задеплоен на `Render` и отлично работает
* Отлично, что фиксируете версии зависимостей
* Отлично, что нет `EOF` ошибок в гите
* Хорошая структура папок и файлов
* `npm audit` показывает **ноль** уязвимостей
*  Проект отлично выглядит и функционирует


но есть некоторые недочеты: 

* В чек-листе есть пункт: Тестами покрыты как минимум шаблонизатор (если делаете свою реализацию), **роутер**, **компонент**, **модуль отправки запросов**. Скрин из чек-листа https://disk.yandex.ru/i/fu8v7hdHCsuluw
* в `pre-commit` нужно еще тесты запускать ( не только линтеры)



**Можно лучше**

* Для линтинга на прекоммите можно воспользоваться библиотекой [lint-staged](https://www.npmjs.com/package/lint-staged) . Она позволяет осуществлять линтинг только тех файлов, которые **изменились**
* `webpack.config.js`  хорошо бы разбить на несколько частей - `development` и `production`. Так же будет еще файл с общими частями для обеих сборок. Объединить конфиги можно с помощью библиотеки `webpack-merge`
* Попробуйте избавиться от типов `any`


Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте **работоспособность проекта** и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

AleksandrMenshchikov committed:
modified:   .husky/pre-commit
	modified:   package-lock.json
	modified:   package.json
	renamed:    webpack.config.js -> webpack.common.js
	new file:   webpack.dev.js
	new file:   webpack.prod.js

AleksandrMenshchikov committed:
modified:   package.json

AleksandrMenshchikov committed:
modified:   package-lock.json
	modified:   package.json

AleksandrMenshchikov committed:
modified:   package.json

AleksandrMenshchikov committed:
modified:   .husky/pre-commit
	modified:   babel.config.js
	new file:   jest.config.js
	modified:   package-lock.json
	modified:   package.json
	new file:   src/core/Router.test.ts
	modified:   src/core/Router.test.ts

AleksandrMenshchikov committed:
modified:   src/core/Router.test.ts

AleksandrMenshchikov committed:
modified:   src/core/Router.test.ts

AleksandrMenshchikov committed:
modified:   babel.config.js
	modified:   jest.config.js
	modified:   package-lock.json
	modified:   package.json
	modified:   src/core/Router.test.ts
	modified:   src/core/Router.ts

AleksandrMenshchikov committed:
modified:   src/core/Router.test.ts

AleksandrMenshchikov committed:
modified:   src/core/Router.test.ts

AleksandrMenshchikov committed:
new file:   src/components/button/button.test.ts

AleksandrMenshchikov committed:
modified:   src/components/button/button.test.ts

AleksandrMenshchikov committed:
new file:   src/tests/setup.ts

AleksandrMenshchikov committed:
modified:   package-lock.json

AleksandrMenshchikov committed:
new file:   src/components/button-add-user/button-add-user.test.ts
	new file:   src/components/button-avatar/button-avatar.test.ts

AleksandrMenshchikov committed:
modified:   src/components/button-avatar/button-avatar.test.ts
	new file:   src/components/button-delete-user/button-delete-user.test.ts

AleksandrMenshchikov committed:
new file:   src/components/button-edit/button-edit.test.ts
	new file:   src/components/button-exit/button-exit.test.ts

AleksandrMenshchikov committed:
new file:   src/components/button-open-profile/button-open-profile.test.ts
	new file:   src/components/button-profile/button-profile.test.ts

AleksandrMenshchikov committed:
new file:   src/components/button-search/button-search.test.ts
	new file:   src/components/button-switch-page/button-switch-page.test.ts

AleksandrMenshchikov committed:
new file:   src/components/clip/clip.test.ts
	new file:   src/components/dots/dots.test.ts
	new file:   src/components/empty-messages/empty-messages.test.ts

AleksandrMenshchikov committed:
new file:   src/components/form-avatar-error/form-avatar-error.test.ts
	new file:   src/components/form-avatar/form-avatar.test.ts

AleksandrMenshchikov committed:
modified:   src/components/form-avatar/form-avatar.test.ts

AleksandrMenshchikov committed:
new file:   src/components/form-edit/form-edit.test.ts
	new file:   src/components/form-login/form-login.test.ts
	new file:   src/components/form-message-error/form-message-error.test.ts

AleksandrMenshchikov committed:
modified:   src/components/form-avatar/form-avatar.test.ts
	modified:   src/components/form-edit/form-edit.test.ts
	modified:   src/components/form-login/form-login.test.ts
	new file:   src/components/form-settings/form-settings.test.ts
	new file:   src/components/form-signup/form-signup.test.ts

AleksandrMenshchikov committed:
new file:   src/components/input-error/input-error.test.ts
	new file:   src/components/input/input.test.ts

AleksandrMenshchikov committed:
new file:   src/components/input-file/input-file.test.ts
	new file:   src/components/input-search/input-search.test.ts
	new file:   src/components/label-file/label-file.test.ts
	new file:   src/components/list-chats/list-chats.test.ts
	new file:   src/components/list-divider/list-divider.test.ts
	new file:   src/components/list-users/list-users.test.ts
	new file:   src/components/loader/loader.test.ts
	new file:   src/components/messages-content/messages-content.test.ts
	new file:   src/components/messages/messages.test.ts

AleksandrMenshchikov committed:
new file:   src/components/modal-add-delete-user/modal-add-delete-user.test.ts
	new file:   src/components/modal-clip/modal-clip.test.ts
	new file:   src/components/modal-edit/modal-edit.test.ts
	new file:   src/components/modal-profile-avatar-title/modal-profile-avatar-title.test.ts
	new file:   src/components/modal-users-title/modal-users-title.test.ts
	new file:   src/components/modal-users/modal-users.test.ts
	new file:   src/components/password-error/password-error.test.ts
	new file:   src/components/profile-arrow/profile-arrow.test.ts
	new file:   src/components/profile-input/profile-input.test.ts
	new file:   src/components/profile-title/profile-title.test.ts

AleksandrMenshchikov committed:
Changes to be committed:
	new file:   src/api/auth-api.test.ts

AleksandrMenshchikov committed:
modified:   src/api/auth-api.test.ts

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились. Все 43 теста проходят успешно

Удачного дальнейшего обучения.
